src/efftester.ml

```diff
@@ -695,8 +779,11 @@ module StaticGenerators = struct
           rs
   ;;
 
+  (* TODO: Include more base types - requires also changing `printer_by_etype` *)
+  let basetype_gen = Gen.oneofl [ Int; Float; String ]
   let eff_gen = Gen.oneofl [ (false, false); (true, false) ]
 
+  (* FIXME: to add option as a base type? *)
```

Comment:
No, I think `option` should not be seen as a base type. Base types are the types whose values cannot be decomposed further: unit, int, bool, string (unless you want to see strings as list/array of characters, but this is not how OCaml does it). Only base types have literals, and they do not have separate intro/elim typing rules.